core/src/main/java/org/opensearch/sql/planner/optimizer/rule/EvalPushDown.java

```diff
@@ -42,6 +51,38 @@ public class EvalPushDown<T extends LogicalPlan> implements Rule<T> {
                 return logicalEval;
               });
 
+  public static final Rule<LogicalSort> PUSH_DOWN_SORT =
+      match(sort(evalCapture()))
+          .apply(
+              (sort, logicalEval) -> {
+                List<LogicalPlan> child = logicalEval.getChild();
+                Map<ReferenceExpression, Expression> evalExpressionMap =
+                    logicalEval.getExpressions().stream()
+                        .collect(Collectors.toMap(Pair::getKey, Pair::getValue));
+                List<Pair<SortOption, Expression>> sortList = sort.getSortList();
+                List<Pair<SortOption, Expression>> newSortList = new ArrayList<>();
+                for (Pair<SortOption, Expression> pair : sortList) {
+                  /*
+                   Narrow down the optimization to only support:
+                   1. The expression in sort and replaced expression are both ReferenceExpression.
+                   2. No internal reference in eval.
+                  */
+                  if (pair.getRight() instanceof ReferenceExpression) {
+                    ReferenceExpression ref = (ReferenceExpression) pair.getRight();
+                    Expression replacedExpr = evalExpressionMap.getOrDefault(ref, ref);
```

Comment:
Is it right to return `ref` as default value?

Reply:
Logically, it's right I think, though the name of `replacedExpr` may be a little bit misleading.  It includes 2 cases:
1. the `ref` is produced by eval operator, then it needs to be replaced
2. the `ref` isn't produced by eval operator, then we don't need to replace it. That's why I use the default value `ref`.

How about changing the name of `replacedExpr` to `newExpr`?